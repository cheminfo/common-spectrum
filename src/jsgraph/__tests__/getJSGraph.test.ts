import { Analysis, JSGraph } from '../..';

test('getJSGraph', () => {
  let analysis = new Analysis();
  expect(analysis.id).toHaveLength(8);

  analysis.pushSpectrum(
    {
      x: {
        data: [1, 2],
        units: 'xUnits',
        label: 'X axis [xUnits]',
      },
      y: {
        data: [3, 4],
        units: 'yUnits',
        label: 'Y axis [yUnits]',
      },
    },
    {
      title: 'My spectrum',
      dataType: 'TGA',
      meta: {
        meta1: 'Meta 1',
        meta2: 'Meta 2',
      },
    },
  );

  const jsgraph = JSGraph.getJSGraph([analysis], {
    normalization: { filters: [{ name: 'multiply', options: { value: 100 } }] },
  });

  delete jsgraph.series[0].name;

  expect(jsgraph).toMatchSnapshot();
});
