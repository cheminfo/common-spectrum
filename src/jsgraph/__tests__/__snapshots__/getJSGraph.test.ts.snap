// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getJSGraph 1`] = `
Object {
  "axes": Object {
    "x": Object {
      "display": true,
      "flipped": false,
      "label": "X axis [xUnits]",
      "unit": "",
    },
    "y": Object {
      "display": true,
      "flipped": false,
      "label": "Y axis",
      "unit": "",
    },
  },
  "series": Array [
    Object {
      "data": Object {
        "x": Array [
          1,
          2,
        ],
        "y": Float64Array [
          300,
          400,
        ],
      },
      "style": Array [
        Object {
          "name": "unselected",
          "style": Object {
            "line": Object {
              "color": "#FFB300FF",
              "dash": 1,
              "width": 1,
            },
          },
        },
        Object {
          "name": "selected",
          "style": Object {
            "line": Object {
              "color": "#FFB300FF",
              "dash": 1,
              "width": 3,
            },
          },
        },
      ],
    },
  ],
}
`;
